packages/core-plugin-api/src/analytics/types.ts

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2021 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Common analytics context attributes.
+ */
+export type CommonAnalyticsContext = {
+  /**
+   * The associated plugin.
+   */
+  pluginId: string;
+
+  /**
+   * The ID of the associated route ref.
+   */
+  routeRef: string;
+
+  /**
+   * The name of the associated extension.
+   */
+  extension: string;
```

Comment:
Now that I'm thinking of it, I'm wondering if the `Partial<>` is even necessary.  All the attributes of `CommonAnalyticsContext` will always be set (because of context default values).  So any consumer of that type should be able to rely on them being set.